File: cpp/1784.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;


class Solution {
public:
    bool checkOnesSegment(string s) {
        int n = s.size();
        int l = 1;
        for (int i = 1; i < n; ++i) {
            if (s[i] != s[i - 1] && s[i] == '1')
                ++l;
        }
        
        return l <= 1 ? true : false;
    }
};

File: python3/1784.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        n = len(s)
        l = 1
        for i in range(1, n):
            if s[i] != s[i - 1] and s[i] == '1':
                l += 1

        return l <= 1
